File: PerfTester-Xamarin/PerfTester-Xamarin/Screens/TestDetails/Cells/TestDetailsCell.cs
using System;

using Foundation;
using PerfTesterXamarin.Models;
using UIKit;

namespace PerfTesterXamarin.Screens.TestDetails.Cells
{
    public partial class TestDetailsCell : UITableViewCell
    {
        public static readonly NSString Key = new NSString("TestDetailsCell");
        public static readonly UINib Nib;

        static TestDetailsCell()
        {
            Nib = UINib.FromName("TestDetailsCell", NSBundle.MainBundle);
        }

        protected TestDetailsCell(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        public void SetupCellWithTest(Test test, int index)
        {
            IndexLabel.Text = String.Format("{0})", index);
            TitleLabel.Text = String.Format("Variant: {0}", test.Parameters[index]);
            if (test.Results != null && index < test.Results.Length && test.Results[index] > 0)
            {
                TimeLabel.Text = String.Format("{0}ms", test.Results[index]);
                TimeLabel.Hidden = false;
            }
            else
            {
                TimeLabel.Text = null;
                TimeLabel.Hidden = true;
            }
        }
    }
}

File: PerfTester-Xcode/PerfTester-Xcode/Screens/TestDetails/Cells/TestDetailsCell.swift
//
//  TestDetailsCell.swift
//  PerfTester-Xcode
//
//  Created by Patryk Średziński on 09.06.2018.
//  Copyright © 2018 private.perftesterxcode. All rights reserved.
//

import UIKit

class TestDetailsCell: UITableViewCell {

    @IBOutlet private var indexLabel: UILabel!
    @IBOutlet private var titleLabel: UILabel!
    @IBOutlet private var timeLabel: UILabel!
    
    func setupCellWithTest(test: Test, index: Int) {
        indexLabel.text = String(format: "%d)", index+1)
        titleLabel.text = String(format: "Variant: %d", Int(test.parameters[index]))
        if index < test.results.count, test.results[index] > 0 {
            timeLabel.text = String(format: "%dms", Int(test.results[index]))
            timeLabel.isHidden = false
        } else {
            timeLabel.text = nil
            timeLabel.isHidden = true
        }
    }
}
